DIPs/DIP1xxx-JS.md

```diff
@@ -0,0 +1,75 @@
+# The Deprecation Process
+
+| Field | Value |
+|-----------------|-----------------------------------------------------------------|
+| DIP:            | xxxx |
+| Review Count:   | 0 |
+| Author:         | Jack Stouffer |
+| Implementation: | N/A |
+| Status:         | Draft |
+
+## Abstract
+
+In order to incrementally improve D or it's standard library, it's often necessary to
+mark features or functions for future removal. This document proposes a standardized
+process for language maintainers to remove public features. This process would be 
+used across DMD, Druntime, and Phobos.
+
+## Rationale
+
+There is general disagreement on the best and/or accepted way to remove public
+features. Each deprecation ends being handled slightly differently depending on
+who's handling the pull requests. Standardizing the process makes sure that
+deprecations are done very publicly and carefully, as to minimize breakage and
+to provide clear fixes for user code.
+
+## Description
+
+The following lays out the rules to follow to deprecate a part of D.
+
+A symbol or feature must not be marked for removal on a specific date, but rather on a
+specific release. This allows users to easily know if upgrading will break their
+code or not.
+
+Both at the time of deprecation and removal, a changelog entry must be made.
+
+### Public Functions, Types, and Modules
+
+The symbol(s) must be marked using the `deprecated` keyword with a message containing
+the planned removal period and/or a pointer to more information pertaining to the
+deprecation. The documentation of the symbol(s) must be updated noting the
+deprecation and removal plan. The documentation should contain information to help
+the users using the symbol(s) transition their code away from the symbol(s).
+
+Users must be given at least four major releases before the deprecated symbols
```

Comment:
We don't follow SemVer at the moment. With the closest approximation (`2.B.C`), bumping `B` is called a minor release :scream: 

However, @MartinNowak plans to change the release process soon. It would mean that the meaning of "release" changes from every two months to six months. So we need to be a bit careful here. OTOH four versions is a good value for now and if Martin's release process change (DIP?) goes through, this can be easily adjusted.
Though, I really like our current bi-monthly releases cycle, it's a good measure of stable and getting new stuff every now and then.